controller/contact_controller.go

```diff
@@ -29,8 +29,14 @@ func PostContantsHandler(w http.ResponseWriter, r *http.Request) {
 
 func GetContantsHandler(w http.ResponseWriter, r *http.Request) {
 
-	data, err := service.GetContact(r.Context())
-
+	searchParam := r.URL.Query().Get("search")
+	var data []dto.Contact
+	var err error
+	if searchParam != "" {
```

Comment:
decision making should not be part of the controller. 
This login should be part of the service.